### PR TITLE
[update][s] t2 - preserve spaces and new lines in descriptions

### PR DESCRIPTION
**Pull request by gavram**
## T2 - Spaces and new lines in dataset descriptions

Spaces and new lines are not showing in dataset descriptions. Added paragraph formatting to preserve spaces and new lines showing. Also removed processing of `dataset.descriptionHtml` because it interfere with `processMarkdown`.

**Before changes:**

spaces_before

**After changes:**

spaces_after
 
## T3 - Keywords section moving
Client wanted that Keyword section to be moved below Metadata. 

**Before changes:**

keywords_before

**After changes:**

keywords

keywords_after

## T10 - Group information
Client wanted that Group information should be visible within Metadata.
metadata_group

## T11 - Click on Resource name sends to Preview
Client wants that click on Resource name begins download instead sends to Preview. Also changed bad download link for `Download` button creating by changing
```
<a class="data-file-type mr-1" href="{{ resource.downloadPath }}">
```
to
```
<a class="data-file-type mr-1" href="{{ resource.path }}">
```
## T14 - Preview button not showing
Client wants that `Preview` button not showing when there is no views for particular dataset.

no_preview_button

## Coloring File types buttons
I added coloring for two file types and corrected not coloring for `GeoJSON`

**Before changes:**
coloring_before

**After changes:**
coloring_after

### CODE REVIEW

File: views/showcase.html [hunk 1]
```diff
@@ -180,6 +168,20 @@ <h2 class="bg-gray-200 px-4 py-3 font-bold mt-4 mb-1">{{__('Metadata')}}</h2>
       </div>
       <!-- Metadata -->
 
+      <!-- Keywords section -->
+        <h2 class="bg-gray-200 px-4 py-3 font-bold mt-4 mb-1">{{__('Keywords')}}</h2>
+        {% if dataset.keywords and dataset.keywords.length !== 0 %}
+          <div class="container">
+            <div class="flex flex-wrap ">
+            {% for t in dataset.keywords %}
+              <a href="/search?q=tags:{{t | safe}}" class="h-5 px-1 py-0 text-primary text-xs border rounded-full border-primary mr-3 mb-3 mt-3 lowercase hover:bg-secondary">{{t | safe}}</a>
```
Comment (anuveyatsu) [inline views/showcase.html]: This is great - the only thing is probably we could increase paddings so that the badges look better.

File: views/showcase.html [hunk 2]
```diff
@@ -164,6 +164,18 @@ <h2 class="bg-gray-200 px-4 py-3 font-bold mt-4 mb-1">{{__('Metadata')}}</h2>
             {{snippets.listify([dataset.license])}}
           </dd>
           {% endif %}
+          {% if dataset.groups %}
+          <dt class="info-list_title">Grupper</dt>
+          <dd class="info-list_description">
+            {% for group in dataset.groups %}
+            <p
+              <span class="colGroup break-words">
+                {{group.title}}
```
Comment (anuveyatsu) [inline views/showcase.html]: I would expect this to be clickable so that I can go to a group's page.

File: views/showcase.html [hunk 3]
```diff
@@ -76,7 +76,7 @@ <h3 class="text-xl font-semibold">
           {{ resource.descriptionHtml | safe }}
         </p>
         <nav>
-          <a class="data-file-type mr-1" href="{{ resource.downloadPath }}">
+          <a class="data-file-type mr-1" href="{{ resource.path }}">
```
Comment (anuveyatsu) [inline views/showcase.html]: I think ODDK is using frontend-v2@v1.0.0 which provides `downloadPath`

File: showcase.html [hunk 4]
```diff
@@ -0,0 +1,295 @@
+{% extends "base.html" %}
```
Comment (anuveyatsu) [inline showcase.html]: This commit looks incorrect - the templates should be located in `views/` directory.

File: views/showcase.html [hunk 5]
```diff
@@ -95,13 +95,17 @@ <h3 class="text-xl font-semibold">
             </dl>
           </a>
           {% endfor %}
+          {% for view in resource.views %}
```
Comment (anuveyatsu) [inline views/showcase.html]: I'm not sure if views are property of a `resource` obj here - I think it's a property of `dataset` obj. Also, I think this should go together with the actual preview.

**Comment by anuveyatsu** [comment]
@gavram any updates here?

File: views/showcase.html (hunk 1 above)
Comment (gavram) [reply views/showcase.html]: I increased paddings and size of keywords badges and now it looks like 
keywords_new

File: views/showcase.html (hunk 2 above)
Comment (gavram) [reply views/showcase.html]: Now Group items are clickable. Click on item create query with that search term on all datasets.
group_clickable

File: showcase.html (hunk 4 above)
Comment (gavram) [reply showcase.html]: I think that I already removed this file. I made mistake while uploading this file.

File: views/showcase.html (hunk 3 above)
Comment (gavram) [reply views/showcase.html]: You are right. This was long time ago, but if I remember well, I tested this in other browsers and noticed that download links not forming well in IE with `downloadPath` parameter. Anyway, I revert this changes as you suggested.

File: views/showcase.html (hunk 5 above)
Comment (gavram) [reply views/showcase.html]: `views` are placed in `resources` in `datapackage.json` file. If we relate `views` to `dataset` object, every `resource` will have `Preview`. May be I'm wrong here.
I also think that this should go with actual preview, but I found some inconsistencies here. Example page is:
https://www.opendata.dk/city-of-vejle/sygefravaer
and `datapackage.json file is:
https://www.opendata.dk/city-of-vejle/sygefravaer/datapackage.json
If you look at this file you can see that there is no `view` for second `resource`, but actually there is a preview for this resource. I can't figure out how this preview is generated without settings from the `datapackage.json` file.

File: views/showcase.html [hunk 6]
```diff
@@ -176,7 +176,7 @@ <h2 class="bg-gray-200 px-4 py-3 font-bold mt-4 mb-1">{{__('Metadata')}}</h2>
             {% for group in dataset.groups %}
             <p
               <span class="colGroup break-words">
-                {{group.title}}
+                <a href="/search?q=groups:{{group.name | safe}} {{ query.q }}" class="text-secondary">{{group.title}}</a>
```
Comment (anuveyatsu) [inline views/showcase.html]: @gavram it is easy to simply link to group page, e.g., `/collections/{{ group.name }}`

File: views/showcase.html [hunk 7]
```diff
@@ -18,7 +18,7 @@
         <a class="breadcrumb_link" href="/search">{{__('Data')}}</a>
       </li>
       <li class="breadcrumb_item">
-        <a class="breadcrumb_link" href="/search/{{ dataset.name }}" aria-current="page">{{ dataset.title or dataset.name }}</a>
+        <a class="breadcrumb_link" href="/{{ dataset.organization.name }}/{{ dataset.name }}" aria-current="page">{{ dataset.title or dataset.name }}</a>
```
Comment (anuveyatsu) [inline views/showcase.html]: I'd recommend using `owner.name` instead of `dataset.organization.name`

File: views/showcase.html [hunk 8]
```diff
@@ -97,7 +97,7 @@ <h3 class="text-xl font-semibold">
             </dl>
           </a>
           {% endfor %}
-          {% for view in resource.views %}
+          {% for view in dataset.resources.views %}
```
Comment (anuveyatsu) [inline views/showcase.html]: This change looks like not related to this commit message.

Comment (gavram) [reply views/showcase.html]: I made error while I'm trying to see impact of using `dataset` instead of `resources`. I fixed this error.

File: views/showcase.html (hunk 6 above)
Comment (gavram) [reply views/showcase.html]: I fixed this as you suggested.

File: views/showcase.html (hunk 7 above)
Comment (gavram) [reply views/showcase.html]: I followed your recommendation and added changes from `dataset.organization.name` to `owner.name`
